int32 ID
int32 numNodes
int32[] path
int32[] countMap
